### PR TITLE
Make sure we parse RealVectorValue correctly.

### LINKED ISSUE #24337: `std::vector<RealVectorValue>` input parameters are not retrieved correctly
## Bug Description
Input parameters of type `std::vector<RealVectorValue>` are not parsed/retrieved correctly. `std::vector<Point>`, however, works as expected.

## Steps to Reproduce
For example, if a user supplies their param as
```
myparam = '5 6 7'
```
then in the code, `getParam<std::vector<RealVectorValue>>("myparam")` gives the following result: {(5,0,0), (6,0,0), (7,0,0)} instead of the expected {(5,6,7)}.

## Impact
Incorrect results, but there is a workaround by using `std::vector<Point>` instead since there is an implicit conversion between `Point` and `RealVectorValue`.

### PR DESCRIPTION
**Pull request by grmnptr**
Closes #24337

### CODE REVIEW

File: test/tests/parser/parse_vector_realvectorvalue/tests [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Tests]
+  design = 'Parser.md'
+  issues = '#24337'
+
+  [parse_vector_value]
+    type = 'RunApp'
+    input = 'parse_vectorvalue.i'
+    requirement = 'The system shall support reading in vectors of VectorValues and Points.'
```
Comment (GiudGiud) [inline test/tests/parser/parse_vector_realvectorvalue/tests:8]:
```suggestion
    requirement = 'The system shall support reading inputs with vectors of vectors of values, as well as vectors of points.'
```

**Review by GiudGiud — COMMENTED** [review]
awesome catch

File: test/tests/parser/parse_vector_realvectorvalue/parse_vectorvalue.i [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  dim = 2
+  nx = 10
+  ny = 10
+[]
+
+[UserObjects]
+  [parsee_tester]
+    type = ReadVectorValue
+    vector_realvv = '0.1 0.2 0.3 0.4 0.5 0.6'
+    vector_point = '0.1 0.2 0.3 0.4 0.5 0.6'
```
Comment (loganharbour) [inline test/tests/parser/parse_vector_realvectorvalue/parse_vectorvalue.i:16]: unused

File: test/tests/parser/parse_vector_realvectorvalue/tests [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Tests]
```
Comment (loganharbour) [inline test/tests/parser/parse_vector_realvectorvalue/tests:1]: should probably test a non-multiple of 3

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: test/tests/parser/parse_vector_realvectorvalue/parse_vectorvalue.i (hunk 2 above)
Comment (GiudGiud) [reply test/tests/parser/parse_vector_realvectorvalue/parse_vectorvalue.i:16]: if unused the test requirement must be amended then

**Comment by joshuahansel** [comment]
Thanks @grmnptr! I'm glad it was a 1-liner! (Other than the test...) Looked it over - nothing more from me.

File: test/tests/parser/parse_vector_realvectorvalue/parse_vectorvalue.i (hunk 2 above)
Comment (grmnptr) [reply test/tests/parser/parse_vector_realvectorvalue/parse_vectorvalue.i:16]: Got it thanks guys!

File: test/tests/parser/parse_vector_realvectorvalue/tests [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  design = 'Parser.md'
+  issues = '#24337'
+
+  [parse_vector_value]
```
Comment (loganharbour) [inline test/tests/parser/parse_vector_realvectorvalue/tests:5]: These requirements still need to be cleaned up a little

It's not really vectors of vectors of values, that's handled by std::vector<std::vector<Real>>

The second exception is too technical

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: test/tests/parser/parse_vector_realvectorvalue/tests (hunk 4 above)
Comment (grmnptr) [reply test/tests/parser/parse_vector_realvectorvalue/tests:5]: Okay, I modified them.

File: test/tests/parser/parse_vector_realvectorvalue/tests [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    type = 'RunApp'
+    input = 'parse_vectorvalue.i'
+    requirement = 'The system shall support reading in a vector of VectorValues.'
```
Comment (GiudGiud) [inline test/tests/parser/parse_vector_realvectorvalue/tests:8]:
```suggestion
    requirement = 'The system shall support reading in a vector of size-3 vectors.'
```
I dont want to mention moose classes in test requirements

File: test/tests/parser/parse_vector_realvectorvalue/tests [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  []
+  [parse_vector_value_error]
+    type = 'RunException'
+    input = 'parse_vectorvalue.i'
+    cli_args = "UserObjects/parse_tester/vector_realvv='0.1 0.2 0.3 0.4 0.5 0.6 0.7'"
+    requirement = "The system shall throw an exception if the given input cannot be parsed into a "
+                  "vector of VectorValues."
```
Comment (GiudGiud) [inline test/tests/parser/parse_vector_realvectorvalue/tests:15]:
```suggestion
    requirement = "The system shall throw an exception if an input to a parameter for vectors of size-3 vectors cannot be parsed."
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
Let's wait till next is fixed

**Review by loganharbour — APPROVED** [review]

**Pull request merged**